retrieve.go

```diff
@@ -9,13 +9,25 @@ import (
 func Get(out interface{}, path string) interface{} {
 	result := get(reflect.ValueOf(out), path)
 
-	if result.Kind() != reflect.Invalid && !result.IsZero() {
+	if result.Kind() != reflect.Invalid && !result.IsZero() && result.CanInterface() {
 		return result.Interface()
 	}
 
 	return nil
 }
 
+func GetAllowZero(out interface{}, path string) interface{} {
```

Comment:
let's avoid using another method and add a functional option to this method instead.

Reply:
@thoas fixed to be in one method with an option allowZero in Get. (fixed relevant tests where required)